src/cs50/sql.py

```diff
@@ -56,9 +57,16 @@ def __init__(self, url, **kwargs):
             if not os.path.isfile(matches.group(1)):
                 raise RuntimeError("not a file: {}".format(matches.group(1)))
 
+        # Record the URL (used in testing)
+        self.url = url
+
         # Create engine, disabling SQLAlchemy's own autocommit mode, raising exception if back end's module not installed
         self._engine = sqlalchemy.create_engine(url, **kwargs).execution_options(autocommit=False)
 
+        # Create a variable to hold the session. If None, autocommit is on.
+        self.Session = orm.sessionmaker(bind=self._engine)
```

Comment:
Session (capital S) is the session class created by calling the sessionmaker() function. self.Session is used multiple times to create _session (little s) which are the actual sessions. The alternative to storing this class is to regenerate it every time execute() is called, which I thought was a little inefficient. Documentation about session creation: [https://docs.sqlalchemy.org/en/13/orm/session_basics.html#getting-a-session](https://docs.sqlalchemy.org/en/13/orm/session_basics.html#getting-a-session)

Reply:
And the _session is stored as well iff we're in manual mode, which is why it's there as well. (Because need the session/transaction to persist between calls to execute().)

Reply:
Oh interesting. In that case, let's call it `self._Session` so that it's conveyed as "private"?

Reply:
Done.